src/Sylius/Bundle/CoreBundle/Validator/Constraints/HasAllPricesDefinedValidator.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Bundle\CoreBundle\Validator\Constraints;
+
+use Sylius\Component\Channel\Repository\ChannelRepositoryInterface;
+use Sylius\Component\Core\Model\ChannelPricingInterface;
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Webmozart\Assert\Assert;
+
+/**
+ * @author Mateusz Zalewski <mateusz.zalewski@lakion.com>
+ */
+final class HasAllPricesDefinedValidator extends ConstraintValidator
+{
+    /**
+     * @var ChannelRepositoryInterface
+     */
+    private $channelRepository;
+
+    /**
+     * @param ChannelRepositoryInterface $channelRepository
+     */
+    public function __construct(ChannelRepositoryInterface $channelRepository)
+    {
+        $this->channelRepository = $channelRepository;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function validate($productVariant, Constraint $constraint)
+    {
+        Assert::isInstanceOf($constraint, HasAllPricesDefined::class);
+
+        $enabledChannels = $this->channelRepository->findBy(['enabled' => true]);
+
+        foreach ($enabledChannels as $channel) {
+            /** @var ChannelPricingInterface $channelPricing */
+            $channelPricing = $productVariant->getChannelPricingForChannel($channel);
+            if (null === $channelPricing || null === $channelPricing->getPrice()) {
+                $this->context->addViolationAt('channelPricings', $constraint->getMessage());
+
+                return;
```

Comment:
I think we shouldn't return on first error, we should provide validation error for every amount that should be fulfilled.